### DISCUSSION TITLE
Does the icu crate support getting a character's name?

### DISCUSSION THREAD

**Discussion by jake-low** [unicode-org/icu4x#5446] [Q&A] [answered]
I'd like to implement the following function:

```rust
fn unicode_name(c: char) -> String {
    unimplemented!()
}

#[cfg(test)]
mod tests {
    use super::unicode_name;
    
    #[test]
    fn test_unicode_name {
        assert_eq!(unicode_name('a'), "LATIN SMALL LETTER A");
        assert_eq!(unicode_name('μ'), "GREEK SMALL LETTER MU");
        assert_eq!(unicode_name('→'), "RIGHTWARDS ARROW");
        assert_eq!(unicode_name('\x08'), "BACKSPACE");
    }
}
```

I was hoping that the `icu` crate had this capability. I see that I can get character properties like bidi class, general category, etc using `icu::properties`, but I couldn't find a way to get the character's (primary) name/description. Is there a way to do this? And if not, is it something that's within the scope of goals for this project eventually?

Thanks in advance for any help.

**Comment by Manishearth** [comment, marked as answer]
I don't think we have this API but yes, having names support in `icu_properties` would be in-scope

**Reply by jake-low** [reply]
Thanks!

**Reply by sffc** [reply]
The main area of research when adding this feature is how to most efficiently represent the data. ICU4C has a file `unames.icu` which I believe contains the character names, but is there an opportunity to do it more efficiently in ICU4X?
